core/proxy.go

```diff
@@ -60,6 +52,14 @@ func NewProxy(hoverfly *Hoverfly) *goproxy.ProxyHttpServer {
 		})
 	}
 
+	proxy.OnRequest(matchesFilter(hoverfly.Cfg.Destination)).
```

Comment:
authentication connect handler was unreachable, only this one was called

Reply:
The handlers are invoked in order it's registered to the proxy, this should fix it. It makes sense to have the auth handler intercept requests first.